src/lib.rs

```diff
@@ -568,6 +568,46 @@ impl<'a> Record<'a> {
 }
 
 /// Metadata about a log message.
+///
+/// # Use
+///
+/// `Metadata` structs are created when users of the library use
+/// logging macros.
+///
+/// They are consumed by implementations of the `Log` trait in the
+/// `enabled` method. 
+///
+/// `Record`s use `Metadata` to determine the log message's severity
+/// and target.
+///
+///
+/// Users should use the `log_enabled!` macro in their code to avoid
+/// constructing expensive log messages.
+///
+/// ## Examples
+///
+/// ```rust
+/// # #[macro_use] 
+/// # extern crate log;
+///
```

Comment:
This will include a blank line at the top of the example code. Please put `#` on this line to hide it.

![selection_053](https://cloud.githubusercontent.com/assets/1940490/26523773/6f5e4354-42d3-11e7-87e3-68097426fe65.png)